### ISSUE TITLE
Spaces

### ISSUE THREAD

**Issue by jakecleary** [tannwestlake/ad-man#4]
Let's discuss how spaces should work, in terms of setup, placement, and querying.

## Setup

**TODO**

## Placement (Registering a space basically)

Something like:

```php
Space::create('header')->with([
    'option' => 'value',
    'option' => 'value',
    'option' => 'value'
])->set();
```

## Querying

Something like:

```php
Advert::get()->where([
    'space' => 'header'
]);
```

I guess we need a set of defaults, but everything should be overwriteable for debugging and customisation purposes.

**Comment by CraigChilds94** [comment]
Yeah that makes sense I guess. What about `Space::place('header')->with(...);` ? Makes more sense to me.

And querying seems pretty standard though would we need to do that? It'd be easier if we could just register a space and any adverts that are assigned to that space will be queried when `Space::place()` is called... Then the options allow us to modify details like which order, template etc is used.

**Comment by jakecleary** [comment]
I supposes seeing as they're just terms it wouldn't make sense to be registering them anyway.

We would just need to call them like you suggested:

```php
Space::get('header')->where([
    'option' => 'value',
    'option' => 'value'
]);
```

... which would just get all ads with the term `header`, plus options.

**Comment by jakecleary** [comment]
Although I think `Advert:get();` makes more sense in terms of the API.

**Comment by CraigChilds94** [comment]
Yeah I understood your initial intention but it didn't make a whole lot of sense with the way it's implemented.

And yeah that makes more sense...

**Comment by CraigChilds94** [comment]
It'd be cool to hook up a backend for analytics so we can generate our heat-map
